src/pytorch_lightning/utilities/types.py

```diff
@@ -49,6 +49,7 @@
 ]
 EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
 _DEVICE = Union[torch.device, str, int]
+MAP_LOCATION = Optional[Union[Callable, _DEVICE, Callable[[_DEVICE], _DEVICE], Dict[_DEVICE, _DEVICE]]]
```

Comment:
Optional should be excluded here, otherwise you can't express that map_location is required.

Reply:
Furthermore, can you explain the annotation
`Callable[[_DEVICE], _DEVICE]`? Usually, the map location takes two input arguments. See https://pytorch.org/docs/stable/generated/torch.load.html?highlight=map_location 

```suggestion
MAP_LOCATION = Optional[Union[Callable, _DEVICE, Dict[_DEVICE, _DEVICE]]]
```

Reply:
@carmocca see above, please

Reply:
reverted to https://github.com/Lightning-AI/lightning/pull/13932/commits/0502a4e9fe2f1b229265de8ae82a129e704f65ee and synced with upstream in https://github.com/Lightning-AI/lightning/pull/13932/commits/82fc4528431fd3412e30782ba64bcc65d2adba38